README.md

```diff
@@ -1 +1,198 @@
+
+```yaml
+cart_pole_controller:
+  ros__parameters:
+    type: joint_trajectory_controller/JointTrajectoryController
+    joints:
+       - slider_to_cart
+    write_op_modes:
+       - slider_to_cart
+    state_publish_rate: 25
+    action_monitor_rate: 20
+    constraints:
+      stopped_velocity_tolerance: 0.05
+      goal_time: 5
+```
+
+#### Setting PID gains
+
+To set the PID gains for a specific joint you need to define them inside `<plugin><ros></plugin></ros>`. Using the generic way of defining parameters with `gazebo_ros`. The name of the parameter correspond the name of the joint followed by a dot and the name of the parameter: `p`, `i`, `d`, `i_clamp_max`, `i_clamp_min` and `antiwindup`.
+
+```xml
+<gazebo>
+  <plugin filename="libgazebo_ros2_control.so" name="gazebo_ros2_control">
+    <ros>
+      <namespace>/</namespace>
+      <parameter name="slider_to_cart.p" type="double">50.0</parameter>
+      <parameter name="slider_to_cart.i" type="double">10.0</parameter>
+      <parameter name="slider_to_cart.d" type="double">15.0</parameter>
+      <parameter name="slider_to_cart.i_clamp_max" type="double">3.0</parameter>
+      <parameter name="slider_to_cart.i_clamp_min" type="double">-3.0</parameter>
+      <parameter name="slider_to_cart.antiwindup" type="bool">false</parameter>
+    </ros>
+    ...
+  </plugins>
+</gazebo>
+```
+
+#### Executing the examples
+
+There are some examples in the `gazebo_ros2_control_demos` package. These examples allow to launch a cart in a 30 meter rail.
+
+![](img/cart.gif)
+
+You can run some of the configuration running the following commands:
+
+```bash
+ros2 launch gazebo_ros2_control_demos cart_example_position_pid.launch.py
+ros2 launch gazebo_ros2_control_demos cart_example_position.launch.py
+ros2 launch gazebo_ros2_control_demos cart_example_velocity.launch.py
+```
+
+Send example commands:
+
+When the Gazebo world is launched you can run some of the following commads to move the cart.
+
+```bash
+ros2 run gazebo_ros2_control_demos example_position
+ros2 run gazebo_ros2_control_demos example_velocity
+```
+
+To get or modify the values of the PID controller you can run the following commands:
+
+```bash
+ros2 param get /gazebo_ros2_control slider_to_cart.p
+ros2 param get /gazebo_ros2_control slider_to_cart.i
+ros2 param get /gazebo_ros2_control slider_to_cart.d
+ros2 param get /gazebo_ros2_control slider_to_cart.i_clamp_max
+ros2 param get /gazebo_ros2_control slider_to_cart.i_clamp_min
+```
+
+```bash
+ros2 param set /gazebo_ros2_control slider_to_cart.p 50.0
+ros2 param set /gazebo_ros2_control slider_to_cart.i 10.0
+ros2 param set /gazebo_ros2_control slider_to_cart.d 15.0
+ros2 param set /gazebo_ros2_control slider_to_cart.i_clamp_max 3.0
+ros2 param set /gazebo_ros2_control slider_to_cart.i_clamp_min -3.0
+```
+
+#### Dependencies
+
+ - gazebo_ros_pkgs
+ - realtime_tools
+ - ros2_control
+ - ros2_controllers
+ - xacro
+
+Step to compile `gazebo_ros2_control`
```

Comment:
How about giving this section its own header? Also:

* It would be good to specify the ROS distro that this works with (which I believe must be `foxy`)
* It's always nice to give users instructions on how to install debian dependencies. So how about adding a `rosdep` call after all the cloning, before building?

Reply:
ping @ahcorde

Reply:
I removed this part of the tutorial in favour of the dockerfile